From b0fd7329c7620f1d88172d99723bcb46346dae5d Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Tue, 2 Feb 2016 19:58:51 +0530
Subject: [PATCH 11/24] usbc/fusb: do not enable vconn by default on connect.

Currently the vconn supply to fusb is by default always enabled
and fusb will enable/disable vconn to port partner based connection
state and pd negotiation. But  fusb chip expects the vconn supply
to the chip also to be enable/disable before enabling/disabling the
vconn to its port partner. This patch removes default vconn
enablement during connect as policy engine enable/disable vconn
when required.

Change-Id: I8391fe764d16ab9c9e2dc1f16d0da8a881e17a36
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-18527
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/465551
---
 drivers/usb/typec/phy-fusb300.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/typec/phy-fusb300.c b/drivers/usb/typec/phy-fusb300.c
index 43489fc..35bc049 100644
--- a/drivers/usb/typec/phy-fusb300.c
+++ b/drivers/usb/typec/phy-fusb300.c
@@ -572,19 +572,15 @@ static int fusb300_setup_cc(struct typec_phy *phy, enum typec_cc_pin cc,
 		val_s1 = FUSB300_SWITCH1_TXCC1;
 		if (phy->state == TYPEC_STATE_ATTACHED_UFP)
 			val |= FUSB300_SWITCH0_PD_CC1_EN;
-		else if (phy->state == TYPEC_STATE_ATTACHED_DFP) {
+		else if (phy->state == TYPEC_STATE_ATTACHED_DFP)
 			val |= FUSB300_SWITCH0_PU_CC1_EN;
-			val |= FUSB300_SWITCH0_VCONN_CC2_EN;
-		}
 	} else if (cc == TYPEC_PIN_CC2) {
 		val |= FUSB300_SWITCH0_MEASURE_CC2;
 		val_s1 = FUSB300_SWITCH1_TXCC2;
 		if (phy->state == TYPEC_STATE_ATTACHED_UFP)
 			val |= FUSB300_SWITCH0_PD_CC2_EN;
-		else if (phy->state == TYPEC_STATE_ATTACHED_DFP) {
+		else if (phy->state == TYPEC_STATE_ATTACHED_DFP)
 			val |= FUSB300_SWITCH0_PU_CC2_EN;
-			val |= FUSB300_SWITCH0_VCONN_CC1_EN;
-		}
 	} else { /* cc removal */
 		goto end;
 	}
@@ -1177,6 +1173,7 @@ static int fusb300_enable_vconn(struct typec_phy *phy, bool en)
 	if (ret)
 		dev_err(phy->dev, "%s: Failed to SWITCH0_REG\n", __func__);
 
+	dev_dbg(phy->dev, "%s: en=%d\n", __func__, en);
 	return ret;
 }
 
@@ -1806,6 +1803,7 @@ static int fusb300_enable_autocrc(struct typec_phy *phy, bool en)
 		goto err;
 
 	chip->process_pd = en;
+	dev_dbg(chip->dev, "%s: en = %x", __func__, en);
 
 	mutex_unlock(&chip->lock);
 	return ret;
-- 
1.9.1

